src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ZipArchiveExTests.cs

```diff
@@ -1,48 +1,75 @@
-﻿using NUnit.Framework;
+using NUnit.Framework;
+using System;
+using System.Collections.Concurrent;
 using System.IO;
 using System.Text;
+using System.Threading;
 using Xamarin.Android.Tasks;
 
 namespace Xamarin.Android.Build.Tests
 {
 	[TestFixture]
 	public class ZipArchiveExTests
 	{
-		string temp;
-		string zip;
+		ConcurrentDictionary<string, string> paths = new ConcurrentDictionary<string, string> ();
+		string root;
 
 		[SetUp]
 		public void SetUp ()
 		{
-			string root = Path.GetDirectoryName (GetType ().Assembly.Location);
-			temp = Path.Combine (root, "temp", TestContext.CurrentContext.Test.Name);
+			root = Path.GetDirectoryName (GetType ().Assembly.Location);
+			string temp = Path.Combine (root, "temp", TestContext.CurrentContext.Test.Name);
```

Comment:
Tests were running in Parallel, and because some used the same `root` folder they tripped up over each other. So I had to rework this a bit so it would work in a parallel environment.